d2bs/kolbot/libs/common/Pather.js

```diff
@@ -861,6 +861,51 @@ ModeLoop:
 
 		return targetArea ? me.area === targetArea : me.area !== preArea;
 	},
+	
+	
+	/*
+	Pather.getAct(targetArea);
+	targetArea - id of the area to enter
+	*/
+	getAct: function getAct(targetArea) {
+		var actNumber,
```

Comment:
I might overlook something in this function, but wouldn't this do the same? and give a more clear idea on what is going on. Or am i overlooking something?

```javascript
    /** @return 0|1|2|3|4|5 */
    getAct: function (targetArea) {
        const areas = [40, 75, 103, 109];

        while(areas.length) {
            if (areas.pop() < targetArea) {
                return areas.length;
            }
        }
        return 0;
    },
```

Reply:
Yeah that was my initial thought too when I looked at another get act function, but I said you know what let me be an optimization kitty. It doesn't make sense if the array is of size 4/5, so ill accept this change

Reply:
the previous one was doing a binary search, but you do have a point. on a small array, it wouldn't really matter.